core/packages/test/scripts/build-binary.sh

```diff
@@ -68,8 +68,22 @@ build_relayer()
     cp $relay_bin "$output_bin_dir"
 }
 
+build_ethereum() {
+    echo "Building geth binary"
+    if [ ! -d "$geth_dir" ] ; then
+        echo "clone go ethereum project to $geth_dir"
+        git clone https://github.com/ethereum/go-ethereum.git $geth_dir
+    fi
+    pushd $geth_dir
+    git checkout tags/$geth_version
+    make geth
+    cp "$geth_dir/build/bin/geth" "$output_bin_dir"
+    popd
+}
+
```

Comment:
Will this re-download and install geth on every `start-services.sh` script run?

Reply:
Is this because NIX does not support `v1.11.2` of `geth` in its `go-ethereum` package?

You should be able to install with `go` instead of checking out and building from source:
```cmd
go install github.com/ethereum/go-ethereum/cmd/geth@v1.11.2
```

Reply:
no, the script here will ignore download if `go-ethereum` directory exist and `make geth` will ignore recompile if binary exist.